app/src/ttrpc.rs

```diff
@@ -36,13 +36,18 @@ pub fn ttrpc_main() {
             )
             .get_matches();
 
+    if !Path::new(DEFAULT_UNIX_SOCKET_DIR).exists() {
+        std::fs::create_dir_all(DEFAULT_UNIX_SOCKET_DIR).expect("Create unix socket dir failed");
+    }
+    let keyprovider_socket_file = format!("{UNIX_SOCKET_PREFIX}{DEFAULT_UNIX_SOCKET_DIR}/{KEYPROVIDER_SOCKET_FILE}");
     let keyprovider_socket = app_matches
         .value_of("KeyProvider ttRPC Unix socket addr")
-        .unwrap_or(DEFAULT_KEYPROVIDER_UNIX_ADDR);
+        .unwrap_or(&keyprovider_socket_file);
```

Comment:
We might directly define two consts `keyprovider_socket_file` and `getresource_socket_file`, by which we can avoid concat the two strings in runtime

Reply:
s.t. I prefer the old way